pkg/osa/types.go

```diff
@@ -0,0 +1,111 @@
+package osa
+
+import (
+	"github.com/ghodss/yaml"
+	"github.com/jim-minter/azure-helm/pkg/api"
+	"github.com/jim-minter/azure-helm/pkg/arm"
+	"github.com/jim-minter/azure-helm/pkg/config"
+	"github.com/jim-minter/azure-helm/pkg/helm"
+	"io/ioutil"
+)
+
+type OSA interface {
+	Validate() error
+	GenerateConfig() (*config.Config, error)
+	GenerateHelm() ([]byte, error)
+	GenerateARM() ([]byte, error)
+	Healthz() error
+}
+
+func NewOSA(newManifest, oldManifest []byte) (OSA, error) {
+	return &osaManager{
+		newManifestBytes: newManifest,
+		oldManifestBytes: oldManifest,
+	}, nil
+}
+
+func NewOSAByPath(newManifest, oldManifest string) (OSA, error) {
+	newBytes, err := ioutil.ReadFile(newManifest)
+	if err != nil {
+		return nil, err
+	}
+
+	osa := &osaManager{
+		newManifestBytes: newBytes,
+	}
+
+	if len(oldManifest) > 0 {
+		oldBytes, err := ioutil.ReadFile(newManifest)
+		if err != nil {
+			return nil, err
+		}
+		osa.oldManifestBytes = oldBytes
+	}
+	return osa, nil
+}
+
+var _ OSA = &osaManager{}
+
+type osaManager struct {
+	newManifestBytes []byte
```

Comment:
I don't love storing new and old things side by side here - the "old" is only relevant for validation.  I think it'll be worse if we do the same for config.  I can get behind storing the internal representation manifest and config here though.

Reply:
the reason I went this route was so that I didn't have to validate in the New*** methods.  If that's part of versioned api loading I think these go away and we only have the internal representations to work with.